### PR TITLE
support configuration domain suffix (#460)

### PR DESCRIPTION
**Pull request by teijinyer**

### CODE REVIEW

File: pkg/ingress/kube/util/util.go [hunk 1]
```diff
@@ -80,7 +81,11 @@ func MessageToGoGoStruct(msg proto.Message) (*types.Struct, error) {
 }
 
 func CreateServiceFQDN(namespace, name string) string {
-	return fmt.Sprintf("%s.%s.svc.%s", name, namespace, DefaultDomainSuffix)
+        domainSuffix := os.Getenv("DOMAIN_SUFFIX")
```
Comment (johnlanni) [inline pkg/ingress/kube/util/util.go:84]: It is more suitable as a global variable to avoid calling os.Getenv every time

**Review by johnlanni — CHANGES_REQUESTED** [review]

File: pkg/ingress/kube/util/util.go [hunk 2]
```diff
@@ -30,6 +31,7 @@ import (
 )
 
 const DefaultDomainSuffix = "cluster.local"
+var domainSuffix := os.Getenv("DOMAIN_SUFFIX")
```
Comment (johnlanni) [inline pkg/ingress/kube/util/util.go:34]:
```suggestion
var domainSuffix = os.Getenv("DOMAIN_SUFFIX")
```

**Review by johnlanni — CHANGES_REQUESTED** [review]

File: pkg/ingress/kube/util/util.go [hunk 3]
```diff
@@ -80,7 +82,10 @@ func MessageToGoGoStruct(msg proto.Message) (*types.Struct, error) {
 }
 
 func CreateServiceFQDN(namespace, name string) string {
-	return fmt.Sprintf("%s.%s.svc.%s", name, namespace, DefaultDomainSuffix)
+        if domainSuffix == "" {
+        	domainSuffix = DefaultDomainSuffix
+        }
+	return fmt.Sprintf("%s.%s.svc.%s", name, namespace, domainSuffix)
```
Comment (CH3CHO) [inline pkg/ingress/kube/util/util.go:88]: Incorrect indentations here.

**Review by CH3CHO — CHANGES_REQUESTED** [review]

File: helm/core/templates/controller-deployment.yaml [hunk 4]
```diff
@@ -202,6 +202,8 @@ spec:
               fieldRef:
                 apiVersion: v1
                 fieldPath: spec.serviceAccountName
+          - name: DOMAIN_SUFFIX
+            value: { { .Values.global.proxy.clusterDomain } }
```
Comment (johnlanni) [inline helm/core/templates/controller-deployment.yaml:206]:
```suggestion
            value: {{ .Values.global.proxy.clusterDomain }}
```

**Review by johnlanni — CHANGES_REQUESTED** [review]

**Review by johnlanni — APPROVED** [review]

**Pull request merged**